### PR TITLE
Interrupt flow updates in broker

### PR DESCRIPTION
**Pull request by mohitc1**
Details in https://github.com/AzureAD/ad-accounts-for-android/pull/2956

In Common
1. Added adders to AuthorizationRequest Builder to allow adding extra query parameters or headers to existing collections.

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/opentelemetry/SpanName.java [hunk 1]
```diff
@@ -67,5 +67,10 @@ public enum SpanName {
     DeleteTransferToken,
     RestoreMsaAccounts,
     OnUpgradeReceiver,
-    UpgradeDeviceRegistration
+    UpgradeDeviceRegistration,
+
+    /**
+     * Span covering interrupt handling using PRT protocol.
+     */
+    PrtHandleInterrupt
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/opentelemetry/SpanName.java]: We already had a boolean to indicate is_interrupt_flow. Is this new span a replacement of that?

Comment (mohitc1) [reply common4j/src/main/com/microsoft/identity/common/java/opentelemetry/SpanName.java]: Removed. Wanted to capture interrupt execution separate but I guess it's not adding reasonable value and would incur cost.

**Review by iamgusain — APPROVED** [review]
:shipit:

File: common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java [hunk 2]
```diff
@@ -231,6 +233,14 @@ public B setExtraQueryParams(List<Map.Entry<String, String>> extraQueryParams) {
             return self();
         }
 
+        public B addExtraQueryParam(@NonNull final Map.Entry<String, String> extraQueryParam) {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java:241]: javadoc on public methods

File: common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java [hunk 3]
```diff
@@ -241,6 +251,14 @@ public B setRequestHeaders(HashMap<String, String> requestHeaders) {
             return self();
         }
 
+        public B addRequestHeader(@NonNull final String key, final String value) {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/AuthorizationRequest.java:265]: javadoc on public methods

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**